### PR TITLE
Tow-64 and Tow-65

### PR DESCRIPTION
**Pull request by RYeeAnn**
Added Post and Comment model

### CODE REVIEW

File: townhall/myapi/models.py [hunk 1]
```diff
@@ -35,4 +37,25 @@ class Organization(models.Model):
...
 
     def __str__(self):
-        return self.name
\ No newline at end of file
+        return self.name
+    
+
+class Post(models.Model):
+    id = models.AutoField(primary_key=True)
+    user = models.ForeignKey(User, on_delete=models.CASCADE)
```
Comment (ansel-rangers11) [inline townhall/myapi/models.py:45]: Appropriate choice on the on_delete setting

File: townhall/myapi/models.py [hunk 2]
```diff
@@ -35,4 +37,25 @@ class Organization(models.Model):
...
+    content = models.TextField()
+    created_at = models.DateTimeField(default=timezone.now)
+
+    def __str__(self):
+        return self.id
+    
+
+class Comment(models.Model):
+    comment_id = models.AutoField(primary_key=True)
```
Comment (ansel-rangers11) [inline townhall/myapi/models.py:54]: for now just make all PK of every model `id`

**Review by ansel-rangers11 — CHANGES_REQUESTED** [review]

**Review by GagenvirGill — APPROVED** [review]

File: townhall/myapi/models.py [hunk 3]
```diff
@@ -35,4 +37,25 @@ class Organization(models.Model):
...
+    
+
+class Post(models.Model):
+    id = models.AutoField(primary_key=True)
+    user = models.ForeignKey(User, on_delete=models.CASCADE)
+    content = models.TextField()
+    created_at = models.DateTimeField(default=timezone.now)
+
+    def __str__(self):
+        return self.id
```
Comment (JordanBandur) [inline townhall/myapi/models.py:50]: The str method should return a string, self.id is an integer. 
consider converting it to a string: `return str(self.id)`

File: townhall/myapi/models.py [hunk 4]
```diff
@@ -35,4 +37,25 @@ class Organization(models.Model):
...
+
+class Comment(models.Model):
+    id = models.AutoField(primary_key=True)
+    user = models.ForeignKey(User, on_delete=models.CASCADE)
+    post = models.ForeignKey(Post, on_delete=models.CASCADE)
+    content = models.TextField()
+    created_at = models.DateTimeField(default=timezone.now)
+
+    def __str__(self):
+        return self.id
```
Comment (JordanBandur) [inline townhall/myapi/models.py:61]: Same as above, consider converting self.id to a string

**Review by JordanBandur — CHANGES_REQUESTED** [review]

**Review by JordanBandur — APPROVED** [review]

**Review by ansel-rangers11 — APPROVED** [review]

**Review by ansel-rangers11 — CHANGES_REQUESTED** [review]
Usually a migration file is generated when making a new model, may need to run `makemigrations` https://docs.djangoproject.com/en/5.1/topics/migrations/

**Review by ansel-rangers11 — CHANGES_REQUESTED** [review]
I think you have to merge main first, then generate a migration since 0006 is already taken in main [0006_remove_organization_description_and_more.py](https://github.com/AtriaCoop/townhallbackend/blob/main/townhall/myapi/migrations/0006_remove_organization_description_and_more.py)

This is an example of a migration conflict, and its quite common to encounter unfortunately, the solution is to always check with main if someone else has merged in a new migration file, in which case, you will have to "regenerate" your migration such that the correct order is followed and there are no conflicting migration. Good learning opportunity here too @RYeeAnn @GagenvirGill @JordanBandur

**Review by ansel-rangers11 — APPROVED** [review]
Looks good, merge migration **should** work

**Pull request merged**